docs/datamap/lucene-datamap-guide.md

```diff
@@ -0,0 +1,180 @@
+       |TEXT_MATCH('name:c10')
+     """.stripMargin).show
+
+  spark.stop
+```
+
+#### DataMap Management
+Lucene DataMap can be created using following DDL
+  ```
+  CREATE DATAMAP [IF NOT EXISTS] datamap_name
+  ON TABLE main_table
+  USING "lucene"
+  DMPROPERTIES ('text_columns'='city, name', ...)
+  ```
+
+DataMap can be dropped using following DDL
+  ```
+  DROP DATAMAP [IF EXISTS] datamap_name
+  ON TABLE main_table
+  ```
+To show all DataMaps created, use:
+  ```
+  SHOW DATAMAP 
+  ON TABLE main_table
+  ```
+It will show all DataMaps created on main table.
+
+
+## Lucene DataMap Introduction
+  Lucene datamap are created as index DataMaps and managed along with main tables by CarbonData.
+  User can create as many lucene datamaps required to improve query performance,
+  provided the storage requirements and loading speeds are acceptable.
+  
+  Once lucene datamaps are created, the indexes generated by lucene will be read for pruning till
+  row level for the filter query by launching a spark datamap job. This pruned data will be read to
+  give the proper and faster result
+    
+  For instance, main table called **sales** which is defined as 
+  
+  ```
+  CREATE TABLE datamap_test (
+    name string,
+    age int,
+    city string,
+    country string)
+  STORED BY 'carbondata'
+  ```
+  
+  User can create Lucene datamap using the Create DataMap DDL
+  
+  ```
+  CREATE DATAMAP dm
+  ON TABLE datamap_test
+  USING "lucene"
+  DMPROPERTIES ('TEXT_COLUMNS' = 'name, country')
+  ```
+
+## Loading data
+When loading data to main table, it checks whether any lucene datamaps are present or not, if it is,
+then lucene index files will be generated for all the text_columns (String Columns) given in
+DMProperties which contains information about the blocklet_id, page_id and row_id and for all the
+data of text_columns. These index files will be written inside a folder named as datamap name inside
+each segment directories.
+
+## Querying data
+As a technique for query acceleration, Lucene indexes cannot be queried directly.
+Queries are to be made on main table. An UDF called TEXT_MATCH is registered in spark session, so
+when a query with TEXT_MATCH() is fired, While doing query planning, TEXT_MATCH will be treated as
+pushed filters. It checks for all the lucene datamaps, and a job is fired for pruning and for each
+blocklet a temporary file will be generated which has information till row level, but prune will
+return blocklets finally.
+
+When query reaches executor side, the temporary files written will be read and bitset groups are
+formed to return the query result.
+
+User can verify whether a query can leverage Lucene datamap or not by executing `EXPLAIN`
+command, which will show the transformed logical plan, and thus user can check whether TEXT_MATCH()
+filter is applied on query or not.
+
+
+## Data Management with pre-aggregate tables
+Once there is lucene datamap is created on the main table, following command on the main
```

Comment:
Once lucene datamap is created on the main table, following command on the main table is not supported: